torax/_src/imas_tools/input/core_profiles.py

```diff
@@ -109,3 +108,90 @@ def profile_conditions_from_IMAS(
       "normalize_n_e_to_nbar": False,
       "v_loop_lcfs": v_loop_lcfs,
   }
+
+
+def plasma_composition_from_IMAS(
+    ids: ids_toplevel.IDSToplevel,
+    t_initial: float | None = None,
+) -> dict[str, Any]:
+  """Returns dict with args for plasma_composition config from a given ids.
+
+  Loading IMAS data for plasma composition should only be used with n_e_ratios
+  and n_e_ratios_Z_eff impurity modes, not with fractions mode. The impurity
+  mode needs to be specified explicitly after loading the IMAS data. In case
+  n_e_ratios_Z_eff is specified, one impurity ratio must be set to None
+  explicitly. In case n_e_ratios is used, Z_eff will simply be ignored.
+  Note that if the ids individual ions properties are not filled, it will not
+  raise an error and just return an empty dict as main_ion and species.
+
+  Args:
+    ids: A core_profiles IDS object. The IDS can contain multiple time slices.
+    t_initial: Initial time used to map the profiles in the dicts. If None the
+      initial time will be the time of the first time slice of the ids. Else all
+      time slices will be shifted such that the first time slice has time =
+      t_initial.
+
+  Returns:
+    The updated fields read from the IDS that can be used to completely or
+    partially fill the `plasma_composition` section of a TORAX `CONFIG`.
+  """
+  profiles_1d, rhon_array, time_array = _get_time_and_radial_arrays(
+      ids, t_initial
+  )
+
+  Z_eff = (
+      time_array,
+      rhon_array,
+      [profile.zeff for profile in profiles_1d],
+  )
+  species = {}
+  ratios = {}
+  for ion in range(len(profiles_1d[0].ion)):
+    try:
+      symbol = str(profiles_1d[0].ion[ion].name)
+    except AttributeError:
+      # Case ids is plasma_profiles in early DDv4 releases.
+      symbol = str(profiles_1d[0].ion[ion].label)
+    if symbol in constants.ION_PROPERTIES_DICT.keys():
+      # Fill impurities
+      if symbol not in constants.MAIN_ION_SYMBOLS:
```

Comment:
Make this an optional parameter to allow users to set if they want to allow other things to be main ions

Also maybe switch to have the positive case first to simplify the control flow